hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapAddIndexMessageTask.java

```diff
@@ -77,6 +79,14 @@ public String getMethodName() {
 
     @Override
     public Object[] getParameters() {
-        return new Object[]{parameters.indexConfig};
+        return new Object[] {parameters.indexConfig};
+    }
+
+    @Override
+    protected void beforeProcess() {
+        if (nodeEngine.getConfig().getMapConfig(parameters.name).getInMemoryFormat() == InMemoryFormat.NATIVE
```

Comment:
Should this check be placed at the operation (`AddIndexOperationFactory`) so that it also works for server side test?

Reply:
Only client side uses `AddIndexOperationFactory`. Member side directly creates a new `AddIndexOperation`. So this check should be implemented inside `AddIndexOperation` if there will be only one. However overwriting `innerBeforeRun()` inside `AddIndexOperation` results in multiple exceptions(more than 500) in different threads. So it's not fast failing anymore.

Reply:
There is a check at constructor: https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/map/impl/operation/AddIndexOperation.java#L52  And you may be able put this check into https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/query/impl/IndexUtils.java#L62 method.

Reply:
Inside https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/query/impl/IndexUtils.java#L62 method, only accessible variables are mapName, IndexConfig and some unrelated static variables, hence it's not possible to get memory format. I need something like nodeEngine or recordStore or HazelcastInstance etc. Also inside the constructor only map name is initialized. Other variables are initialized in this https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperation.java#L80-L105 method. However this method is called for each partition so it's throwing exceptions for each partition even if we do the check immediately after we can access the memory type.